Simplenote/src/main/java/com/automattic/simplenote/utils/AccountVerificationWatcher.java

```diff
@@ -29,33 +27,44 @@
     }
 
     private final Simplenote simplenote;
-    private final List<VerificationStateListener> listeners = new ArrayList<>();
+    private final VerificationStateListener listener;
+    private Status currentState = Status.UNKNOWN;
 
-    public AccountVerificationWatcher(Simplenote simplenote) {
+    public AccountVerificationWatcher(Simplenote simplenote, VerificationStateListener listener) {
         this.simplenote = simplenote;
+        this.listener = listener;
     }
 
-    public void addVerificationStateListener(VerificationStateListener listener) {
-        listeners.add(listener);
+    private void notifyListener(Status status) {
+        if (listener != null) {
+            listener.onUpdate(status);
+        }
     }
 
-    private void notifyListeners(Status status) {
-        for (VerificationStateListener listener : listeners) {
-            listener.onUpdate(status);
+    private void updateState(Status newState) {
+        if (newState != currentState) {
+            currentState = newState;
+            notifyListener(newState);
         }
     }
 
+    private boolean isValidChangeType(Bucket.ChangeType type, String key) {
+        return type == Bucket.ChangeType.INDEX ||
+                (type == Bucket.ChangeType.INSERT && KEY_EMAIL_VERIFICATION.equals(key)) ||
+                (type == Bucket.ChangeType.MODIFY && KEY_EMAIL_VERIFICATION.equals(key));
+
+    }
```

Comment:
is it really "valid" that we care about? all changes are "valid" because they are coming from the Simperium library. maybe we're looking for something like `affectsVerification` or `involvesVerification`?

Reply:
I would say that is valid inside the context of `AccountVerificationWatcher`.